cloud/organization/organization_token.go

```diff
@@ -158,35 +227,356 @@ func getOrganizationToken(id, name, message string, tokens []astrocore.ApiToken)
+		return user.ErrNoShortName
+	}
 
-	return APITokens, nil
+	var token astrocore.ApiToken
+	if id == "" {
+		tokens, err := getOrganizationTokens(client)
+		if err != nil {
+			return err
+		}
+		token, err = getOrganizationToken(id, name, "\nPlease select the Organization API token you would like to update:", tokens)
+		if err != nil {
+			return err
+		}
+	} else {
+		token, err = getOrganizationTokenByID(id, ctx.OrganizationShortName, client)
+		if err != nil {
+			return err
+		}
+	}
+
+	apiTokenID := token.Id
+
+	UpdateOrganizationAPITokenRequest := astrocore.UpdateOrganizationApiTokenJSONRequestBody{}
+
+	if newName == "" {
+		UpdateOrganizationAPITokenRequest.Name = token.Name
+	} else {
+		UpdateOrganizationAPITokenRequest.Name = newName
+	}
+
+	if description == "" {
+		UpdateOrganizationAPITokenRequest.Description = token.Description
+	} else {
+		UpdateOrganizationAPITokenRequest.Description = description
+	}
+
+	var currentOrgRole string
+	apiTokenWorkspaceRoles := []astrocore.ApiTokenWorkspaceRole{}
+
+	for i := range token.Roles {
+		if token.Roles[i].EntityType == workspaceEntity {
+			apiTokenWorkspaceRoles = append(apiTokenWorkspaceRoles, astrocore.ApiTokenWorkspaceRole{
+				EntityId: token.Roles[i].EntityId,
+				Role:     token.Roles[i].Role,
+			})
+		}
+
+		if token.Roles[i].EntityType == organizationEntity {
+			currentOrgRole = token.Roles[i].Role
+		}
+	}
+	if role == "" {
+		updateOrganizationAPITokenRoles := astrocore.UpdateOrganizationApiTokenRoles{
+			Organization: currentOrgRole,
+			Workspace:    &apiTokenWorkspaceRoles,
+		}
+
+		UpdateOrganizationAPITokenRequest.Roles = updateOrganizationAPITokenRoles
+	} else {
+		err := user.IsOrganizationRoleValid(role)
+		if err != nil {
+			return err
+		}
+		updateOrganizationAPITokenRoles := astrocore.UpdateOrganizationApiTokenRoles{
+			Organization: role,
+			Workspace:    &apiTokenWorkspaceRoles,
+		}
+		UpdateOrganizationAPITokenRequest.Roles = updateOrganizationAPITokenRoles
+	}
+
+	resp, err := client.UpdateOrganizationApiTokenWithResponse(httpContext.Background(), ctx.OrganizationShortName, apiTokenID, UpdateOrganizationAPITokenRequest)
+	if err != nil {
+		return err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Astro Organization API token %s was successfully updated\n", token.Name)
+	return nil
+}
+
+// rotate a organization API token
+func RotateToken(id, name string, cleanOutput, force bool, out io.Writer, client astrocore.CoreClient) error {
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return user.ErrNoShortName
+	}
+
+	var token astrocore.ApiToken
+
+	if id == "" {
+		tokens, err := getOrganizationTokens(client)
+		if err != nil {
+			return err
+		}
+		token, err = getOrganizationToken(id, name, "\nPlease select the Organization API token you would like to add to the Workspace:", tokens)
```

Comment:
TODO this looks like wrong text likely ill need to fix it in the place it was copied from as well